src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -887,6 +888,7 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
     ).subscribe((questionToCopyCandidate: QuestionToCopyCandidate[]) => {
       questionToCopyCandidates.push(...questionToCopyCandidate);
     }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); }, () => {
+      this.isCopyingQuestion = false;
```

Comment:
Why `isCopyingQuestion` is set only when there is error message?

Reply:
It's not the error callback, it's in the `complete` callback. See https://angular.io/guide/observables#defining-observers

Reply:
I don't think `complete` will be called after `error`? https://stackoverflow.com/questions/33783967/rxjs-observable-doesnt-complete-when-an-error-occurs

Think `finalize` is better.

Reply:
But then we run into the problem you raised previously about there being additional loading before the modal pops up. That loading is the copying within the subscription and finalise is called at the end of the pipe instead of at the end of the subscription operation. 

You're also right though that complete won't be called on error, so I'll put in the toggle into the error block as well.

Reply:
The additional loading I mentioned was due to the fetching of the intents. Your old implementation works, just that you need to put the `finalize` after `flatMap`. See https://stackoverflow.com/questions/54567073/finalize-with-flatmap-in-angular.